src/TitleDetails/TitleDetails.js

```diff
@@ -148,7 +148,7 @@ const TitleDetails = ({
   const numberOfPhysicalUnits = useMemo(() => {
     return poLine?.locations?.reduce((acc, { quantityPhysical = 0 }) => acc + quantityPhysical, 0);
   }, [poLine?.locations]);
-  const vendor = vendorsMap[order.vendor];
+  const vendor = vendorsMap[order?.vendor];
```

Comment:
The same here - `order` is the required prop. It seems bad data comes from the outside

Reply:
We should prevent the page to be breaking